nixos/modules/config/xdg/autostart.nix

```diff
@@ -1,26 +1,40 @@
-{ config, lib, ... }:
+{ config, lib, pkgs, ... }:
 
-with lib;
+let
+  cfg = config.xdg.autostart;
+in
 {
   meta = {
-    maintainers = teams.freedesktop.members;
+    maintainers = lib.teams.freedesktop.members;
   };
 
   options = {
-    xdg.autostart.enable = mkOption {
-      type = types.bool;
+    xdg.autostart.enable = lib.mkOption {
+      type = lib.types.bool;
       default = true;
       description = lib.mdDoc ''
         Whether to install files to support the
         [XDG Autostart specification](https://specifications.freedesktop.org/autostart-spec/autostart-spec-latest.html).
       '';
     };
+    xdg.autostart.packages = lib.mkOption {
+      type = with lib.types; listOf package;
```

Comment:
So all of them will be added... I thought the user can solve that with a wrapper. Or it's better to use the path to the desktop files directly?

Reply:
Thinking about it some more, I would say creating new autostart files should be out of scope of the autostart module.

To reduce maintenance burden, I would like to narrow down various freedesktop modules to only handle the core functionality, which here is just making sure autostart files are available where XDG-compliant launchers expect them.

I think the ideal API for this would be something like:

```nix
environment.systemPackages = [
  (pkgs.mkAutostartFileFromLauncher pkgs.firefox "firefox")
];
```

Reply:
Ah, I didn't know that there is such a function. :)